core/src/main/java/org/elasticsearch/client/transport/TransportClientNodesService.java

```diff
@@ -523,6 +518,7 @@ public String executor() {
                                     public void handleResponse(ClusterStateResponse response) {
                                         clusterStateResponses.put(nodeToPing, response);
                                         latch.countDown();
+                                        closeConnection();
```

Comment:
maybe we should unify the `latch.countDown()` and `closeConnection()` into a single method called "onDone" on the AbstractRunnable that everyone calls? this it's less trappy and people wouldn't forget to do one but not the other.